spirv_glsl.cpp

```diff
@@ -11813,11 +11868,52 @@ void CompilerGLSL::emit_instruction(const Instruction &instruction)
 
 	// ALU
 	case OpIsNan:
-		GLSL_UFOP(isnan);
+		if (!is_legacy())
+			GLSL_UFOP(isnan);
+		else if (expression_type(ops[2]).vecsize > 1)
+			// Check if the number doesn't equal itself
+			emit_binary_func_op(ops[0], ops[1], ops[2], ops[2], "notEqual");
+		else
+			emit_binary_op(ops[0], ops[1], ops[2], ops[2], "!=");
 		break;
 
 	case OpIsInf:
-		GLSL_UFOP(isinf);
+		if (!is_legacy())
+			GLSL_UFOP(isinf);
+		else
+		{
+			// inf * 2 == inf by IEEE 754 rules, note this also applies to 0.0
```

Comment:
A bit questionable without precise, but since this is only for legacy targets, sure.